ssc/cmod_wavefile.cpp

```diff
@@ -60,100 +78,255 @@ class cm_wave_file_reader : public compute_module
+                }
+                month[r] = (ssc_number_t)std::stod(values[month_index]);
+                timecheck[r] = (ssc_number_t)std::stod(values[hour_index]);
+                if (r > 0) {
+                    if (timecheck[r] - timecheck[r - 1] != hourdiff && timecheck[r] != 0) {
+                        throw exec_error("wave_file_reader", "Time steps are nonuniform");
+                        timecheck[r] = 999;
+                    }
+                }
+                p_year[r] = (ssc_number_t)std::stod(values[year_index]);
+                p_hour[r] = (ssc_number_t)std::stod(values[hour_index]);
+                p_day[r] = (ssc_number_t)std::stod(values[day_index]);
+                p_minute[r] = (ssc_number_t)std::stod(values[minute_index]);
+                wave_heights[r] = (ssc_number_t)std::stod(values[height_index]);
+                wave_periods[r] = (ssc_number_t)std::stod(values[period_index]);
+
+                //Make JPD from time series data
+
+                ts_significant_wave_height = wave_heights[r];
+                ts_energy_period = wave_periods[r];
+                for (size_t j = 0; j < 21; j++) {
+
+                    if (r == 0) mat[(j + 1) * ncols] = (0.25 + j * 0.5);
+                    if (abs(ts_significant_wave_height - (0.25 + (j - 1) * 0.5)) <= 0.25 && j != 0) {
+                        sig_wave_height_index = j;
+                        if (r != 0) break;
+                    }
+                }
+                for (size_t m = 0; m < 22; m++) {
+                    if (r == 0) mat[m] = m - 0.5;
+                    if (abs(ts_energy_period - (0.5 + (m - 1))) <= 0.5 && m != 0) {
+                        energy_period_index = m;
+                        if (r != 0) break;
+                    }
+
+
+                }
+
+
+                //mat[sig_wave_height_index * ncols + energy_period_index] = mat[sig_wave_height_index * ncols + energy_period_index] + 1 / 2920 * 100;
+                mat[sig_wave_height_index * ncols + energy_period_index] += 0.0342465753;
+                //Set decimal values to 2 for JPD
+                if (r == numberRecords - 1) {
+                    for (size_t r2 = 0; r2 < 21; r2++) {
+                        for (size_t c2 = 0; c2 < 22; c2++) {
+                            if (r2 != 0 && c2 != 0) mat[r2 * 22 + c2] = round(mat[r2 * 22 + c2] * 100) / 100;
+                        }
+                    }
+                }
+                
+            }
+            mat[0] = 0;
+            assign("number_hours", int(numberRecords * hourdiff));
+            return;
+
+        }
+        else if (as_integer("wave_resource_model_choice") == 0) {
+            ssc_number_t* mat = allocate("wave_resource_matrix", 21, 22);
+            for (size_t r = 0; r < 21; r++)
+            {
+                getline(ifs, buf);
+                values.clear();
+                values = split(buf);
+                if (values.size() != 22)
```

Comment:
Throw an error, and give a more descriptive error message for why there should be 22 columns. What is the label and units on this column?